packages/orchestration/src/service.js

```diff
@@ -97,78 +124,136 @@ const prepareOrchestrationKit = (
-        async allocateICQControllerPort() {
-          const portAllocator = getPower(this.state.powers, 'portAllocator');
-          return E(portAllocator).allocateICAControllerPort();
+      },
+      requestICQConnectionWatcher: {
+        /**
+         * @param {Port} port
+         * @param {{
+         *   remoteConnAddr: RemoteIbcAddress;
+         *   controllerConnectionId: IBCConnectionID;
+         * }} watchContext
+         */
+        onFulfilled(port, { remoteConnAddr, controllerConnectionId }) {
+          const connectionKit = makeICQConnectionKit(port);
+          /** @param {ICQConnectionKit} kit */
+          return watch(
+            E(port).connect(remoteConnAddr, connectionKit.connectionHandler),
+            this.facets.channelOpenWatcher,
+            {
+              connectionKit,
+              returnFacet: 'connection',
+              saveICQConnection: controllerConnectionId,
+            },
+          );
         },
       },
+      /**
+       * Waits for a channel (ICA, ICQ) to open and returns the consumer-facing
+       * facet of the ConnectionKit, specified by `returnFacet`. Saves the
+       * ConnectionKit if `saveICQConnection` is provided.
+       */
+      channelOpenWatcher: {
+        /**
+         * @param {Connection} _connection
+         * @param {{
+         *   connectionKit: ConnectionKit;
+         *   returnFacet: string;
+         *   saveICQConnection?: IBCConnectionID;
+         * }} watchContext
+         */
+        onFulfilled(
+          _connection,
+          { connectionKit, returnFacet, saveICQConnection },
+        ) {
+          if (saveICQConnection) {
+            this.state.icqConnections.init(
+              saveICQConnection,
+              /** @type {ICQConnectionKit} */ (connectionKit),
+            );
+          }
+          return connectionKit[returnFacet];
+        },
+        // TODO #9317 if we fail, should we revoke the port (if it was created in this flow)?
+        // onRejected() {}
+      },
       public: {
         /**
          * @param {IBCConnectionID} hostConnectionId the counterparty
          *   connection_id
          * @param {IBCConnectionID} controllerConnectionId self connection_id
          * @returns {Promise<IcaAccount>}
          */
-        async makeAccount(hostConnectionId, controllerConnectionId) {
-          const port = await this.facets.self.allocateICAControllerPort();
-
+        makeAccount(hostConnectionId, controllerConnectionId) {
           const remoteConnAddr = makeICAChannelAddress(
             hostConnectionId,
             controllerConnectionId,
           );
-          const chainAccountKit = makeChainAccountKit(port, remoteConnAddr);
-
-          // await so we do not return a ChainAccount before it successfully instantiates
-          await E(port).connect(
-            remoteConnAddr,
-            chainAccountKit.connectionHandler,
+          const portAllocator = getPower(this.state.powers, 'portAllocator');
```

Comment:
There's a comment here: https://github.com/Agoric/agoric-sdk/blob/19bc874e26b3257ccf48e93bd454153e5ca8be88/packages/orchestration/src/service.js#L29-L31

Reply:
thanks, that makes sense. I was looking on getPower.

So that explains why a store instead of a record, and I take it the getter is because maps assume homogeneity of value types. That could be worthwhile to include as a comment.

Reply:
btw, going forward s/7337/7407 https://github.com/Agoric/agoric-sdk/issues/7337#issuecomment-2150419761

Reply:
we talked about getPower and how to get rid of it, right, @0xpatrickdev ? (IOU link)
we don't need to complicate access to known state fields to provide for expansion.
we just need one reserved-for-future-use property

Reply:
- https://github.com/Agoric/agoric-sdk/pull/9353

Reply:
Thanks @dckc . I'll tackle this in a follow PR